app/grandchallenge/core/utils/markdown.py

```diff
@@ -41,3 +44,72 @@ def run(self, root):
             if el.tag == "a":
                 el.set("target", "_blank")
                 el.set("rel", "noopener")
+
+
+class EmbedYoutubeExtension(Extension):
+    def extendMarkdown(self, md):  # noqa: N802
+        md.registerExtension(self)
+        md.inlinePatterns.register(
+            EmbedYouTubeInLineProcessor(
+                r"^\[\s*youtube\s*([A-Za-z0-9_-]{11})\s*([0-9]*)\s*([0-9]*)\s*\]",
+                md,
+            ),
+            "embed_youtube_extension",
+            0,
+        )
+
+
+class EmbedYouTubeInLineProcessor(InlineProcessor):
+    def handleMatch(self, m, data):  # noqa: N802
+        youtube_id = m.group(1)
+
+        el = ElementTree.Element("iframe")
+
+        el.set(
```

Comment:
Don't love this way of constructing a HTML fragment. Could you use a Django partial template instead and then load it as an element? Find those much easier to follow.